### DISCUSSION TITLE
feat: Add alternative way to create procedures

### DISCUSSION THREAD

**Discussion by KATT** [trpc/trpc#2998] [RFC / Ideas]
### Describe the feature you'd like to request

Add a simpler alternative way to define procedures.

This is the current way of doing it:

```ts
import { publicProcedure, router } from './trpc';
import { z } from 'zod';
 
export const appRouter = router({
  hello: publicProcedure
    .input(
      z
        .object({
          text: z.string(),
        })
        .optional(),
    )
    .query(({ input }) => {
      return {
        greeting: `hello ${input?.text ?? 'world'}`,
      };
    }),
});

```

### Describe the solution you'd like to see

Most of the time, I use some sort of base procedure that does all the authz/authn, and end up just doing `.input(x).query(() => { /*...*/} )` etc at the end.

Here I'm proposing to **adding** an alternative way of defining procedures that I think is a bit more readable:

```ts
import { publicProcedure, router } from './trpc';
import { z } from 'zod';

export const appRouter = router({
  hello: publicProcedure.query({
    input: z
      .object({
        text: z.string(),
      })
      .optional(),
    resolve({ input }) {
      return {
        greeting: `hello ${input?.text ?? 'world'}`,
      };
    },
  }),
});
```

### Desribe alternate solutions

- There's maybe some alternative way of addressing this
- This is very much a nice-to-have. 
- Maybe it can be a bit misleading to have two ways of doing things

### Additional information

- Related: #2940
- `input` should not be required
- Needs to work when the base procedures have existing input parsers
- If someone wants to pick this up, I'll happily provide a test suite to work against

### 👨‍👧‍👦 Contributing

- [X] 🙋‍♂️ Yes, I'd be down to file a PR implementing this feature!

**Comment by sachinraja** [comment]
Both ways are just as readable IMO. I'd prefer we be consistent. People are bound to be confused if we offer multiple ways of doing the same thing. We're also already maintaining a large API surface.
